src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -80,41 +79,35 @@ object XSDToSchema {
                 restriction.getBaseTypeName
               }
             matchType match {
-              case Constants.XSD_BOOLEAN => StructField(baseName, BooleanType)
-              case Constants.XSD_BYTE => StructField(baseName, BinaryType)
-              case Constants.XSD_DATE |
```

Comment:
Looks like you removed Date/DaateTime? You could release this with https://github.com/databricks/spark-xml/pull/448

Reply:
I just let them fall into the general string case. But yeah maybe just as well to try them as a date or time type.